mne/decoding/transformer.py

```diff
@@ -245,6 +246,111 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(TransformerMixin):
+    """Transforms n-dimensional array into 2D array of n_samples by n_features.
+
+    This class reshapes an n-dimensional array into an n_samples * n_features
+    array, usable by the estimators and transformers of scikit-learn.
+
+    Examples
+    --------
+    clf = make_pipeline(SpatialFilter(), XdawnTransformer(), Vectorizer(),
+                        LogisticRegression())
+
+    Attributes
+    ----------
+    ``features_shape_`` : tuple
+         Stores the original shape of data.
+    """
+
+    def fit(self, X, y=None):
+        """Stores the shape of the features of X.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to fit. Can be, for example a list, or an array of at
+            least 2d. The first dimension must be of length n_samples, where
+            samples are the independent samples used by the estimator
+            (e.g. n_epochs for epoched data).
+        y : None
+            Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        self : Instance of Vectorizer
+            Return the modified instance.
+        """
+        X = np.asarray(X)
+        self.features_shape_ = X.shape[1:]
+        return self
+
+    def transform(self, X):
+        """Convert given array into two dimensions.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to fit. Can be, for example a list, or an array of at
+            least 2d. The first dimension must be of length n_samples, where
+            samples are the independent samples used by the estimator
+            (e.g. n_epochs for epoched data).
+
+        Returns
+        -------
+        X : array, shape (n_trials, -1)
+            The transformed data.
+        """
+        X = np.asarray(X)
+        if X.shape[1:] != self.features_shape_:
+            raise ValueError("Shape of X used in fit and transform must be "
+                             "same")
+        return X.reshape(len(X), -1)
+
+    def fit_transform(self, X, y=None):
+        """Fit the data, then transform in one step.
+
+        Parameters
+        ----------
+        X : array-like
+            The data to fit. Can be, for example a list, or an array of at
+            least 2d. The first dimension must be of length n_samples, where
+            samples are the independent samples used by the estimator
+            (e.g. n_epochs for epoched data).
+        y : None
+            Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        X : array, shape (n_trials, -1)
+            The transformed data.
+        """
+        return self.fit(X).transform(X)
+
+    def inverse_transform(self, X):
+        """Transform 2D data back to its original feature shape.
+
+        Parameters
+        ----------
+        X : array-like
```

Comment:
I think this docstring is wrong. For inverse_transform X is 2D (n_samples, n_features)

please check your docstrings